### DISCUSSION TITLE
O que vocês acham importante/essencial aprender sobre React?

### DISCUSSION THREAD

**Discussion by synga** [frontendbr/forum#1870] [Discussão]
Fala pessoas. Estou começando a estudar React agora pra dar uma ampliada nas opções, e é uma lib que venho acompanhando faz um tempo, mas só o que uma pessoa ou outra fala sobre, nunca peguei a fundo pra entender as opções e tudo que ela oferece.

Sendo assim queria saber o que acham relevante, importante ou essencial de se estudar com ou sobre React agora em 2020, seja em questão de CSS, testes, padrões, qualquer coisa.
Tirando o Redux e o React Native que já estão na minha lista, queria saber o que mais seria legal pra poder montar um roadmap aqui de estudos.

**Comment by ninetails** [comment]
Partindo da premissa que já entenda de JavaScript _baunilha_... E o que eu vou falar está mais para um plano de estudos que algo prático. Mas o prático vc consegue comprando um curso na Udemy com os instrutores explicando bem :wink: (não to fazendo propaganda não, o curso pode ser qualquer um... também não estou falando que qualquer curso vai ser bom, mas aí o assunto é outro e não é o foco da pergunta rs)

Antes de tudo, entender a própria _lib_. O porquê que todo mundo insiste que ela é uma biblioteca e não framework, e que o que seria "framework" é todo o ecossistema em volta que vc tem a liberdade de utilizar a alternativa que quiser no desenvolvimento (da mesma forma, quando vc entrar num projeto React, cada projeto vai ter sua "stack", suas personalizações decididas por quem desenvolveu anteriormente).

Também é interessante saber Node.js pois atualmente todos os grandes frameworks utilizam alguma forma de compilação. Como desconheço qual o teu nível, se já utilizou algum outro framework atual pra frontend, estará _safe_ aqui.

Outra coisa é que cada "framework" novo tem sua personalidade, essência. (Estarei falando bem por cima aqui...) Vue é focado em algo mais prático e com menor curva de conhecimento. Angular foca em estrutura de dados... Com o React não é diferente, ele teve inspiração em bibliotecas funcionais e sua base é no Virtual DOM. Não precisa conhecer a fundo, mas é legal entender como funciona.

Também tem a questão do famoso **JSX** que galera torce o nariz quando vê pela primeira vez. Não tem nada muito místico, é entender que vc está escrevendo ECMAScript com uma outra sintaxe, mais declarativa.

Depois de pegar mais confiança, separar responsabilidades... O que seria um componente lógico, o que seria um componente presentacional, componentes funcionais e algumas funcionalidades da nova versão como Hooks e Context API. E lá na frente, tipagem... recomendo TypeScript.

Em relação a testes, tudo o que vc encontrar do [Kent C. Dodds](https://github.com/kentcdodds) é bem pertinente e atual.

Estilos, é importante saber **CSS** (claro), mas também como importá-los... Os principais modos são: importar direto, importar utilizando _css modules_ e _CSS-in-JS_. Claro que para entender, talvez seja necessário também aprender um pouco sobre _Bundlers_, sendo os mais utilizados: [Webpack](https://webpack.js.org/) e [Rollup](https://rollupjs.org/), mas também tem alguns mais novos e fáceis de usar como o [Parcel](https://parceljs.org/).

Padrões para escrita de código, o [ESLint](https://eslint.org/) possui [plugin pra React](https://github.com/yannickcr/eslint-plugin-react).

### Enfim, precisa saber tudo isso? E isso é tudo é necessário para começar em React?

Por favor, não. Como vc falou que quer fazer um _Roadmap_ de estudos, só quis colocar alguns pontos pra ficar de olho durante os estudos. Tem diversos cursos que vc pode aprender na prática e mais rápido. O que botei acima são coisas pra estudar quando "tiver tempo", que podem te agregar conhecimento complementar e entender mais quando estiver fazendo algum curso.

**Comment by synga** [comment]
@ninetails adorei as dicas, muito do que você falou nem tinha pensado em tentar entender. Já comecei um curso na Udemy e já vou incluir esses tópicos nos estudos. Muito obrigado
